### DISCUSSION TITLE
Best resource for Flutter ...

### DISCUSSION THREAD

**Discussion by SjxSubham** [community/community#128084] [Programming Help] [answered]
What are the best resources to learn flutter and the roadmap..

**Comment by davevad93** [comment, marked as answer]
Hi @SjxSubham ,

Here you can find the best online resources to learn `Flutter`:

1. [Flutter documentation](https://docs.flutter.dev/), in my opinion the must do thing to learn any programming language is to look to their official documentation.
2. [Introduction to Flutter](https://www.udemy.com/course/introduction-to-flutter/), this is a Udemy free course which teaches you the basics about `Flutter`.
3. [freeCodeCamp](https://www.freecodecamp.org/news/search/?query=Flutter), here you can find a lot of projects to practice in `Flutter` and develop your apps.
4. [Codelabs](https://docs.flutter.dev/codelabs), the `Flutter` codelabs provide a guided, hands-on coding experience.

And here's a complete roadmap to become a proper `Flutter` developer: [Flutter Developer](https://roadmap.sh/flutter)

Hope this helps!
